src/components/BookPlace.tsx

```diff
@@ -1,173 +1,179 @@
 import AddPayment from './AddPayment'
 import addYears from '../modules/addYears'
+import BookingPlaceData from '../interfaces/BookingPlaceData'
+import FiniteDateRange from '../interfaces/FiniteDateRange'
+import CustomDateRange from '../interfaces/CustomDateRange'
+import Button from './Button'
+import InputGroup from './InputGroup'
+import Checkbox from './Checkbox'
+import Dialog from './Dialog'
+import createFirebaseTimestampFromDate  from '../modules/createFirebaseTimestampFromDate'
 
 interface BookPlaceProps {
   placeId: string
-  handleClose: () => void
+  onClose: () => void
   placeBookings: BookingPlaceData[]
   placeName: string
 }
 
-const BookPlace: React.FC<BookPlaceProps> = ({ handleClose, placeId, placeBookings, placeName }) => {
+const BookPlace: React.FC<BookPlaceProps> = ({ onClose, placeId, placeBookings, placeName }) => {
   // Getting info from inputs
   const [visitorName, setVisitorName] = useState<string>('')
-  const handleNameChange = (event: React.ChangeEvent<HTMLInputElement>) => setVisitorName(event.target.value)
+  const onNameChange = (event: React.ChangeEvent<HTMLInputElement>) => setVisitorName(event.target.value)
 
   const [amount, setAmount] = useState<number>()
 
-  const [isFormsDisabled, setIsFormsDisabled] = useState(false)
+  const [disabledFlag, setDisabledFlag] = useState(false)
 
-  const sendPayment = (amount: number, paidDays: FiniteDateRange) => {
+  const setPayment = (amount: number, paidDays: FiniteDateRange) => {
     setAmount(amount)
     setPaidDays(paidDays)
   }
 
   // Getting and sending data with firebase
   const { book } = useBookingPlace(placeId)
-  const bookPlace = (event: React.FormEvent<HTMLFormElement>) => {
+  const onSubmit = (event: React.FormEvent<HTMLFormElement>) => {
     event.preventDefault()
-    setIsFormsDisabled(true)
-
-    if (bookedDays && paidDays && visitorName && amount) {
-      book(
-        {
-          startDate: Timestamp.fromDate(bookedDays.startDate),
-          endDate: bookedDays.endDate ? Timestamp.fromDate(bookedDays.endDate) : null,
-          visitorName,
-          placeName,
-          amount,
-          paidDays: {
-            startDate: Timestamp.fromDate(paidDays.startDate),
-            endDate: Timestamp.fromDate(paidDays.endDate)
-          }
-        }, placeBookings
-      )
-      .then(() => {
-        AppToaster.show({ message: 'The place has been booked' })
-        setIsFormsDisabled(false)
-        handleClose()
-      })
-      .catch(({ message }) => {
-        AppToaster.show({ message, intent: 'danger' })
-        setIsFormsDisabled(false)
-      })
+    setDisabledFlag(true)
+
+    if (!bookedDays || !paidDays || !visitorName || !amount) {
+      setDisabledFlag(false)
+      return
     }
+    else bookPlace(amount, paidDays, bookedDays)
+  }
+
+  const bookPlace = (amount: number, paidDays: FiniteDateRange, bookedDays: CustomDateRange) => {
+    if (!bookedDays) return
+    book(
+      {
```

Comment:
```
const newBooking = { ... };
book(newBooking, placeBookings)
```

Reply:
Let's do the validation in the `bookPlace`, instead of passing `placeBookings` down to the `book` function.